### PR TITLE
Added caching for `display` property of `Location` and `LocationType`…

### PR DESCRIPTION
**Pull request by Kircheneer**
# Closes: #4573 
# What's Changed

Backports some of the changes #3702 to `ltm/1.6` in order to improve performance for the location/location type API endpoints.

# TODO
- [x] Explanation of Change(s)
- [x] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

**Review by glennmatthews — COMMENTED** [review]
Don't suppose you can test this in the 100k-Location deployment and let us know how much it helps? :-)

File: changes/4573.added [hunk 1]
```diff
@@ -0,0 +1 @@
+Added caching for `display` property of `Location` and `LocationType`, mitigating duplicated in the related API views.
```
[the remark this answers is not in the document]
Comment (Kircheneer) [reply changes/4573.added]: Fixed, thanks!

File: nautobot/utilities/tree_queries.py [hunk 2]
```diff
@@ -17,3 +18,37 @@ class TreeManager(Manager.from_queryset(TreeQuerySet), TreeManager_):
...
+        """
+        if not hasattr(self, "name"):
+            raise NotImplementedError("default TreeModel.display implementation requires a `name` attribute!")
+        cache_key = f"{self.__class__.__name__}.{self.id}.display"
+        display_str = cache.get(cache_key, "")
+        if display_str:
+            return display_str
+        try:
+            if self.parent is not None:
+                display_str = self.parent.display + " → "
```
[the remark this answers is not in the document]
Comment (Kircheneer) [reply nautobot/utilities/tree_queries.py:47]: As discussed verbally, the access pattern over the API view doesn't exhibit a query pattern where this optimization is useful as it pulls in the parent through `select_related`. As such I have no real way to test this and we won't deviate from the code in the 2.x branch(es).

**Comment by Kircheneer** [comment]
I have tested this and it dramatically improves performance for all API endpoints related to locations.

**Review by jathanism — APPROVED** [review]
:shipit:

**Review by glennmatthews — APPROVED** [review]

**Pull request merged**